### PR TITLE
V2 Split files

### PR DESCRIPTION
**Pull request by frack113**
Split information in smaller file

- [appendix_modifier](appendix_modifer.md) is a document that defines the different modifier use that can be used in a Sigma rule. 
- [appendix_tags](appendix_tags.md) is a document that defines the standardized tags that can be used to categorize the different Sigma rules.
- [appendix_taxonomy](appendix_taxonomy.md) is a document that defines the different field names and log sources that should be used to ensure sharable rules

### CODE REVIEW

File: V2_breaking_changes.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+
+Warning `sigmac` will not be able to convert this version. Only `pySigma` and the corresponding `sigma-cli` provide full support for version 2.
+
+
+# New Modifier
+
+- `windash` : converts `-` values into `/` and vice versa. Will be used for all `CommandLine` fields in windows > `process_creation` rules.
+- `exists` that allows to define that a certain field exists or doesn't exists in a log event by providing
+  a boolean value. Currently we use filters with `field: null` as a workaround for this purpose, which is inaccurate.
```
Comment (phantinuss) [inline V2_breaking_changes.md:11]:
```suggestion
  a boolean value. Currently we use filters with `field: null` as a workaround for this purpose, which is less descriptive.

```

File: V2_breaking_changes.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+# Corelation
+
+- Remove aggregation expression in sigma rule file see [Sigma meta rules](Sigma_meta_rules.md)
```
Comment (phantinuss) [inline V2_breaking_changes.md:15]:
```suggestion
- Remove aggregation expression in Sigma rule file see [Sigma meta rules](Sigma_meta_rules.md)
```

File: appendix_modifer.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Modifiers <!-- omit in toc -->
+
+The following document defines the standardized modifiers that can be used in the Sigma rules.
```
Comment (phantinuss) [inline appendix_modifer.md:3]:
```suggestion
The following document defines the standardized modifiers that can be used in Sigma.

```

File: appendix_modifer.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+
+## Transformations
+
+* `contains`: puts `*` wildcards around the values, such that the value is matched anywhere in the
+  field.
+* `all`: Normally, lists of values were linked with *OR* in the generated query. This modifier
+  changes
+  this to *AND*. This is useful if you want to express a command line invocation with different
+  parameters where the order may vary and removes the need for some cumbersome workarounds.
```
Comment (phantinuss) [inline appendix_modifer.md:22]:
```suggestion
* `all`: Normally, lists of values are linked with *OR* in the generated query. This modifier
  changes this to *AND*. This is useful if you want to express a command line invocation with different
  parameters where the order may vary and removes the need for some cumbersome workarounds.
```

File: appendix_modifer.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  might change depending on the position of the value in the overall string. There are three variants for shifts
+  by zero to two bytes and except the first and last byte the encoded values have a static part in
+  the middle that can be recognized.
+* `endswith`: The value is expected at the end of the field's content (replaces e.g. '*\cmd.exe')
+* `startswith`: The value is expected at the beginning of the field's content. (replaces e.g. 'adm*')
+* `utf16le`: Transforms value to UTF16-LE encoding, e.g. `cmd` > `63 00 6d 00 64 00` (only used in combination with base64 modifiers)
+* `utf16be`: Transforms value to UTF16-BE encoding, e.g. `cmd` > `00 63 00 6d 00 64` (only used in combination with base64 modifiers)
+* `wide`: Alias for `utf16le` modifier
+* `utf16`: Prepends a [byte order mark](https://en.wikipedia.org/wiki/Byte_order_mark) and encodes UTF16, e.g. `cmd` > `FF FE 63 00 6d 00 64 00` (only used in combination with base64 modifiers)
+* `windash`: Add a new variant where all `-` occurrences are replaced with `/`. The original variant is also kept unchanged.
```
Comment (phantinuss) [inline appendix_modifer.md:37]: Todo for me: Elaborate on `windash` here

File: appendix_modifer.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+* `cidr`: The value is handled as an IPv4 CIDR by backends (IPv6 is not supported, yet)
```
Comment (phantinuss) [inline appendix_modifer.md:38]:
```suggestion
* `cidr`: The value is handled as an CIDR by backends
```
waiting for verification by Thomas

File: appendix_modifer.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+* `lt`: Field is less than the value
+* `lte`: Field is less or egal than the value
+* `gt`: Field is Greater than the value
+* `gte`: Field is Greater or egal than the value
```
Comment (phantinuss) [inline appendix_modifer.md:42]:
```suggestion
* `lte`: Field is less or equal than the value
* `gt`: Field is Greater than the value
* `gte`: Field is Greater or equal than the value

```

File: appendix_modifer.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+* `expand`: Modifier for expansion of placeholders in values. It replaces placeholder strings
```
Comment (phantinuss) [inline appendix_modifer.md:43]:
```suggestion
* `expand`: Modifier for expansion of placeholders in values. It replaces placeholder strings. (e.g. ..?)
```
This one needs an example or more explanation, I think. E.g. how is a string expanded if the input is a list, like names of admin users.

File: appendix_modifer.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+## Types
+
+* `re`: value is handled as regular expression by backends. Currently, this is only supported by
+  the Elasticsearch query string backend (*es-qs*). Further (like Splunk) are planned or have
+  to be implemented by contributors with access to the target systems.
```
Comment (phantinuss) [inline appendix_modifer.md:49]:
```suggestion
* `re`: value is handled as a regular expression by backends.
```
Is this even true? In any way I think this shouldn't be part of the spec but be a part of each backends Readme to tell the user which features of the standard are not supported

File: appendix_modifer.md (hunk 6 above)
Comment (thomaspatzke) [reply appendix_modifer.md:38]: Confirmed, pySigma supports both IP versions.

File: appendix_modifer.md (hunk 8 above)
Comment (thomaspatzke) [reply appendix_modifer.md:43]: It's currently documented [here](https://sigmahq-pysigma.readthedocs.io/en/latest/Processing_Pipelines.html#sigma.processing.transformations.WildcardPlaceholderTransformation), a blog post on this is on my todo list as one of the next action points!

File: appendix_modifer.md (hunk 9 above)
Comment (thomaspatzke) [reply appendix_modifer.md:49]: Agree! We should remove any backend-specific information from the spec.

At least the Splunk pySigma backend also supports this and it's part of the base class and cookiecutter template, so everyone who writes a backend gets at least a chance to implement this 😉

File: README.md [hunk 10]
```diff
@@ -15,10 +15,11 @@ The version number is in the form of 3 digits 'A.B.C':
 The Sigma rules format specifications is described in the file [Sigma_specification](Sigma_specification.md)  
 The Sigma Meta-rules format specifications is described in the file [Sigma_meta_rules](Sigma_meta_rules.md)  
 
-There exists two other files in the repository to describe the different fields and tags to be used in Sigma rules:
+There exists other files in the repository to describe the different modifier, fields and tags to be used in Sigma rules:
```
[the remark this answers is not in the document]
Comment (thomaspatzke) [reply README.md:18]: Shouldn't the "exists" not be a singular "exist"?

**Review by thomaspatzke — APPROVED** [review]
Great work, @frack113 & @phantinuss!

File: appendix_modifer.md [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+  might change depending on the position of the value in the overall string. There are three variants for shifts
+  by zero to two bytes and except the first and last byte the encoded values have a static part in
+  the middle that can be recognized.
+* `endswith`: The value is expected at the end of the field's content (replaces e.g. '*\cmd.exe')
+* `startswith`: The value is expected at the beginning of the field's content. (replaces e.g. 'adm*')
+* `utf16le`: Transforms value to UTF16-LE encoding, e.g. `cmd` > `63 00 6d 00 64 00` (only used in combination with base64 modifiers)
+* `utf16be`: Transforms value to UTF16-BE encoding, e.g. `cmd` > `00 63 00 6d 00 64` (only used in combination with base64 modifiers)
+* `wide`: Alias for `utf16le` modifier
+* `utf16`: Prepends a [byte order mark](https://en.wikipedia.org/wiki/Byte_order_mark) and encodes UTF16, e.g. `cmd` > `FF FE 63 00 6d 00 64 00` (only used in combination with base64 modifiers)
+* `windash`: Add a new variant where all `-` occurrences are replaced with `/`. The original variant is also kept unchanged.
```
Comment (phantinuss) [inline appendix_modifer.md:37]:
```suggestion
* `windash`: creates all possible permutations of the `-` and `/` characters. Windows command line flags can often be indicated by both characters. Using the `windash` modifier converts `-` values into `/` and vice versa and uses all possible permutation of strings in the selection.

```

**Review by phantinuss — APPROVED** [review]

File: appendix_modifer.md [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+* `gte`: Field is Greater or equal than the value
+
+* `expand`: Modifier for expansion of placeholders in values. The final behavior of the replacement is determined by processing pipeline transformations. Current possibilities in pySigma are:
+  * Expand to value list (`ValueListPlaceholderTransformation`/`value_placeholders`)
+  * Replace with query expression in target query language (`QueryExpressionPlaceholderTransformation`/`query_expression_placeholders`)
+  * Replace placeholder with wildcard `*`, which should only be used as last resort. (`WildcardPlaceholderTransformation`/`wildcard_placeholders`)
+
+## Types
+
+* `re`: value is handled as a regular expression by backends.
```
Comment (phantinuss) [inline appendix_modifer.md:52]: weren't there more value modifiers in context of `|re`.

Also the `|cased`.

Comment (frack113) [reply appendix_modifer.md]: I did not work on the content of the files, I separated them at first

**Pull request merged**